src/source-pfring.c

```diff
@@ -283,6 +287,42 @@ static inline void PfringProcessPacket(void *user, struct pfring_pkthdr *h, Pack
     SET_PKT_LEN(p, h->caplen);
 }
 
+#ifdef PF_RING_FLOW_OFFLOAD
+/**
+ * \brief Pfring bypass callback function
+ *
+ * \param p a Packet to use information from to trigger bypass
+ * \return 1 if bypass is successful, 0 if not
+ */
+static int PfringBypassCallback(Packet *p)
+{
+    hw_filtering_rule r;
+
+    /* Only bypass TCP and UDP */
+    if (!(PKT_IS_TCP(p) || PKT_IS_UDP(p))) {
+        return 0;
+    }
+
+    /* Bypassing tunneled packets is currently not supported */
+    if (IS_TUNNEL_PKT(p)) {
+        return 0;
+    }
+
+    r.rule_family_type = generic_flow_id_rule;
+    r.rule_family.flow_id_rule.action = flow_drop_rule;
+    r.rule_family.flow_id_rule.thread = 0;
+    r.rule_family.flow_id_rule.flow_id = p->pfring_v.flow_id;
+
+    SCLogDebug("Bypass set for flow ID = %u", p->pfring_v.flow_id);
+
+    if (pfring_add_hw_rule(p->pfring_v.ptv->pd, &r) < 0) {
```

Comment:
what does it mean if this failed? Should we report failure somehow? Perhaps in a counter?

Reply:
If for some reason it fails (e.g. hw resources are exhausted) it should fall back to local bypass. Not sure if we need to report it.

Reply:
I guess that depends on whether the user can somehow act on this. E.g. assign more resources. If it's purely about hardware limitations then there is not much to be done probably.

Reply:
With the cards we have today with offload support, it's a pure hw limitation, no user action is needed nor helpful

Reply:
ok

Another point: shouldn't this func be checked for in configure.ac?

Reply:
It's under #ifdef PF_RING_FLOW_OFFLOAD, do you think it's not enough?

Reply:
I think it would be better to add it to configure.ac so that configure output will show if it's available or not. This will help us support users that try to use it.